### PR TITLE
Feature/hand listener

### PR DESCRIPTION
**Pull request by davidpn11**
Adding: 
- listen to changes on his cards
- update from the activeCard

### CODE REVIEW

**Comment by davidpn11** [comment]
Shitty description because Michel was pissing me

File: src/api/db/gameSession.ts [hunk 1]
```diff
@@ -20,3 +23,25 @@ export async function requestGetPlayerHand(
...
+  callback: (h: string[]) => void,
+) {
+  try {
+    await getSessionRef(sessionId)
+      .collection('players')
+      .doc(currentPlayerId)
+      .onSnapshot(documentSnapshot => {
+        const player = extractDocumentData<SessionPlayer>(documentSnapshot);
+
+        if (O.isSome(player)) {
```
Comment (mikelfcosta) [inline src/api/db/gameSession.ts]: You should do a fold an throw an error if there is no player cause that is an impossible state.

File: src/api/db/preGameSession.ts [hunk 2]
```diff
@@ -199,17 +199,44 @@ export const requestBuyCards = (sessionRef: ReturnType<typeof getSessionRef>) =>
 async function requestSetCurrentCard(sessionRef: ReturnType<typeof getSessionRef>): Promise<Card> {
   const deckRef = sessionRef.collection('deck');
   const deck = normalizeQuery<Card>(await deckRef.get());
-  const userCards = popDeckCards(deck, 'GAME');
-  const key = pipe(userCards.keys, A.head);
+
+  let currentCard = popDeckCards(deck, 'GAME');
+
+  while (
```
Comment (mikelfcosta) [inline src/api/db/preGameSession.ts:205]: Damn

File: src/api/db/preGameSession.ts [hunk 3]
```diff
@@ -199,17 +199,44 @@ export const requestBuyCards = (sessionRef: ReturnType<typeof getSessionRef>) =>
...
   const card = pipe(
     key,
     O.fold(
       () => O.none,
-      key => R.lookup(key, userCards.cards),
+      key => R.lookup(key, currentCard.cards),
     ),
   );
 
+  console.log(card);
```
Comment (mikelfcosta) [inline src/api/db/preGameSession.ts]: log

**Review by mikelfcosta — APPROVED** [review]

**Pull request merged**